drf_rw_serializers/generics.py

```diff
@@ -75,16 +70,21 @@ def get_read_serializer(self, *args, **kwargs):
         kwargs["context"] = self.get_serializer_context()
         return serializer_class(*args, **kwargs)
 
-    def get_read_serializer_class(self):
+    def get_read_serializer_class(self, use_serializer_class: bool = False):
         """
         Return the class to use for the serializer.
         Defaults to using `self.read_serializer_class`.
+
         You may want to override this if you need to provide different
         serializations depending on the incoming request.
+
         (Eg. admins get full serialization, others get basic serialization)
         """
         if getattr(self, "read_serializer_class", None) is None:
-            return self._get_serializer_class()
+            if use_serializer_class:
+                return self.serializer_class
+
+            return self.get_serializer_class()
```

Comment:
That's the approach I found to avoid raising a `RecursionError` while allowing:
1. `get_serializer_class` refer to `get_read_serializer_class` and `get_write_serializer_class` 
2. `get_read_serializer_class` and `get_write_serializer_class` refer back to `get_serializer_class`

Reply:
Hi @pamella @fjsj ,

Okay guys, apparently your `RecursionError` fix will break potentially everyone who override `get_read_serializer_class` or `get_write_serializer_class`.

```
TypeError: MyViewSet.get_read_serializer_class() got an unexpected keyword argument 'default_to_serializer_class'
```

When we override before, we never include any arguments like:
```python
@typing_extensions.override
def get_read_serializer_class(self) -> drf_serializers.BaseSerializer[MyModel]:
    if self.action == 'my_action`:
        return MySerializer
    else:
        return super().get_read_serializer_class()
```

With this change everything become broken... I do not like the fix, also I would like to understand why do we have this "infinite recursion"? In version `1.1.1` everything was working and this workaround was not needed. What has changed since then?

Reply:
> _In version 1.1.1 everything was working and this workaround was not needed. What has changed since then?_

@sshishov Please check the main PR that implemented the breaking changes, [Update get_serializer_class to consider the request method #77](https://github.com/vintasoftware/drf-rw-serializers/pull/77). Its description contains more information about the issues it was meant to address.

Reply:
@sshishov you need to accept *args, **kwargs in your `def get_read_serializer_class`. We'll update the Usage docs to mention this. Unfortunately we see no other way of supporting the browsable API w/o the changes introduced in this PR #87.

```python
@typing_extensions.override
def get_read_serializer_class(self, *args, **kwargs) -> drf_serializers.BaseSerializer[MyModel]:
    if self.action == 'my_action`:
        return MySerializer
    else:
        return super().get_read_serializer_class()
```